process/block/shardblock.go

```diff
@@ -761,13 +775,19 @@ func (sp *shardProcessor) getHighestHdrForShardFromMetachain(shardId uint32, hdr
 
 		ownHdr, err := process.GetShardHeader(shardInfo.HeaderHash, sp.dataPool.Headers(), sp.marshalizer, sp.store)
 		if err != nil {
-			continue
+			go sp.onRequestHeaderHandler(shardInfo.ShardId, shardInfo.HeaderHash)
```

Comment:
ask for all your headers, and return error after.

Reply:
done